src/Pact/Types/Term/Internal.hs

```diff
@@ -0,0 +1,1039 @@
+  {-# INLINEABLE toJSON #-}
+  {-# INLINEABLE toEncoding #-}
+
+instance FromJSON Use where
+  parseJSON = withObject "Use" $ \o ->
+    Use <$> o .: "module"
+        <*> o .:? "hash"
+        <*> o .:? "imports"
+        <*> o .: "i"
+
+instance NFData Use
+instance SizeOf Use
+
+-- -------------------------------------------------------------------------- --
+-- Guard
+
+data Guard a
+  = GPact !PactGuard
+  | GKeySet !KeySet
+  | GKeySetRef !KeySetName
+  | GModule !ModuleGuard
+  | GUser !(UserGuard a)
+  deriving (Eq,Show,Generic,Functor,Foldable,Traversable,Ord)
+
+instance NFData a => NFData (Guard a)
+
+instance Pretty a => Pretty (Guard a) where
+  pretty (GPact g)      = pretty g
+  pretty (GKeySet g)    = pretty g
+  pretty (GKeySetRef g) = pretty g
+  pretty (GUser g)      = pretty g
+  pretty (GModule g)    = pretty g
+
+instance (SizeOf p) => SizeOf (Guard p) where
+  sizeOf (GPact pg) = (constructorCost 1) + (sizeOf pg)
+  sizeOf (GKeySet ks) = (constructorCost 1) + (sizeOf ks)
+  sizeOf (GKeySetRef ksr) = (constructorCost 1) + (sizeOf ksr)
+  sizeOf (GModule mg) = (constructorCost 1) + (sizeOf mg)
+  sizeOf (GUser ug) = (constructorCost 1) + (sizeOf ug)
+
+keyNamef :: AesonKey
+keyNamef = "keysetref"
+
+instance ToJSON a => ToJSON (Guard a) where
+  toJSON (GKeySet k) = enableToJSON "Pact.Types.Term.Guard a" $ toJSON k
+  toJSON (GKeySetRef n) = enableToJSON "Pact.Types.Term.Guard a" $ object [ keyNamef .= n ]
+  toJSON (GPact g) = enableToJSON "Pact.Types.Term.Guard a" $ toJSON g
+  toJSON (GModule g) = enableToJSON "Pact.Types.Term.Guard a" $ toJSON g
+  toJSON (GUser g) = enableToJSON "Pact.Types.Term.Guard a" $ toJSON g
+
+  toEncoding (GKeySet k) = toEncoding k
+  toEncoding (GKeySetRef n) = A.pairs (keyNamef .= n)
+  toEncoding (GPact g) = toEncoding g
+  toEncoding (GModule g) = toEncoding g
+  toEncoding (GUser g) = toEncoding g
+
+  {-# INLINEABLE toJSON #-}
+  {-# INLINEABLE toEncoding #-}
+
+instance FromJSON a => FromJSON (Guard a) where
+  parseJSON v =
+      (GKeySet <$> parseJSON v) <|>
+      (withObject "KeySetRef" $ \o -> GKeySetRef <$> o .: keyNamef) v <|>
+      (GPact <$> parseJSON v) <|>
+      (GModule <$> parseJSON v) <|>
+      (GUser <$> parseJSON v)
+  {-# INLINEABLE parseJSON #-}
+
+-- -------------------------------------------------------------------------- --
+-- Module
+
+data Module g = Module
+  { _mName :: !ModuleName
+  , _mGovernance :: !(Governance g)
+  , _mMeta :: !Meta
+  , _mCode :: !Code
+  , _mHash :: !ModuleHash
+  , _mBlessed :: !(HS.HashSet ModuleHash)
+  , _mInterfaces :: ![ModuleName]
+  , _mImports :: ![Use]
+  } deriving (Eq,Functor,Foldable,Traversable,Show,Generic)
+
+instance NFData g => NFData (Module g)
+
+instance Pretty g => Pretty (Module g) where
+  pretty Module{..} = parensSep
+    [ "module" , pretty _mName , pretty _mGovernance , pretty _mHash ]
+
+moduleProperties :: ToJSON g => JsonProperties (Module g)
+moduleProperties o =
+  [ "hash" .= _mHash o
```

Comment:
What decided this particular ordering of the members?

Reply:
The order that is induced by the use of `HashMap` in JSON encoding with old versions of `hashable` and `unordered-containers` and that is codified on chain.